src/modules/booking/booking.service.ts

```diff
@@ -110,4 +108,107 @@ export class BookingService {
       throw new BadRequestException('Booking dates cannot be in the past');
     }
   }
+
+  async getBookingById(id: number): Promise<GetBookingResponse> {
+    const booking = await this.prismaService.booking.findUnique({
+      where: { id },
+    });
+
+    if (!booking) {
+      throw new NotFoundException('Booking not found');
+    }
+
+    return booking;
+  }
+
+  async getBookingsByUserId(userId: number): Promise<GetBookingResponse[]> {
+    const bookings = await this.prismaService.booking.findMany({
+      where: { userId },
+    });
+
+    if (!bookings || bookings.length === 0) {
+      throw new NotFoundException('No bookings found for the user');
+    }
+
+    return bookings;
+  }
+
+  async updateBooking(
+    id: number,
+    userId: number,
+    newBookingData: UpdateBookingDto,
+  ): Promise<CreateBookingResponse> {
```

Comment:
this function is okay, but I would try to refactor this. It is big

Reply:
maybe you could do like this:

data.room = getNewRoomUpdateClause(...)
data.totalPrice = getNewTotalPrice(...)

or something like this. Try to ask Chatgpt